src/app/queries/providers.ts

```diff
@@ -19,6 +52,66 @@ export const useProvidersQuery = (): QueryResult<IProvidersByType> => {
 
   return sortIndexedResultsByName<Provider, IProvidersByType>(result);
 };
+export const useCreateProvider = () => {
+  const client = useClientInstance();
+  const useProviderPost = async (
+    values: OpenshiftFormState['values'] | VMwareFormState['values']
+  ) => {
+    const provider: INewProvider = convertFormValuesToProvider(values);
+    useCheckIfResourceExists(
+      client,
+      VirtResourceKind.Provider,
+      providerResource,
+      provider.metadata.name
+    );
+
+    try {
+      const secret: INewSecret | undefined = convertFormValuesToSecret(
+        values,
+        VirtResourceKind.Provider
+      );
+      if (secret) {
+        await client.create(secretResource, secret);
+      }
+
+      const providerResult = await client.create(providerResource, provider);
+
+      return providerResult;
+    } catch (error) {
+      console.error('Failed to add provider.');
+      return Promise.reject(error);
+    }
+  };
+
+  const [mutate, { isIdle, isLoading, isError, isSuccess, data, error }] = useMutation(
+    useProviderPost,
+    {
+      onSuccess: () => {
+        console.log('did we succeed');
+        queryCache.invalidateQueries('providers');
+      },
+      onError: () => {
+        console.log('did we fail');
+      },
+    }
+  );
+
+  const createProvider = async (formValues) => {
+    try {
```

Comment:
I think maybe the `try/catch` here isn't necessary, I'm not sure but I think any exceptions would be caught inside the post function above, and then `onError` would be called instead of an exception making it to this point